### ISSUE TITLE
no-fallthrough triggers on Linux build agent but not locally on OS X

### ISSUE THREAD

**Issue by sephie** [eslint/eslint#5093]
I'm using ESLint v2.0.0-beta.2 and I found a bug. I used this code

```javascript
//SWITCH STATEMENT
//CASE
//...
  else {
    //...
    return false;
}
//break; //ESLint2beta2: unreachable code
case KEY_CODE.PLUS: //ESLint2beta2: no-fallthrough
case KEY_CODE.PLUSNUMERICPAD:
            var inputElement = this;
//...
```
and this configuration
```
"parser": "babel-eslint",
"extends": "eslint:recommended",
```
And got this output:
*Local (OS X):*
unreachable code triggered (correctly). fixed by commenting `break;`
no errors after that

*Build agent (Linux): *
`build	28-Jan-2016 15:06:19	  127:9  error  Expected a "break" statement before "case"  no-fallthrough`

I expected this instead:
Same output on both environments (either a fallthrough or unreachable code for this code)

**Comment by eslintbot** [comment]
@sephie Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by btmills** [comment]
@sephie what happens if you use the default parser instead of babel-eslint?

**Comment by nzakas** [comment]
Is it possible you're using different versions of ESLint (or babel-eslint) on your computer?

**Comment by sephie** [comment]
No change as far as the fallthrough issue is concerned.

Locally, the default parser equally complains about the unreachable code
After removing the offending unreachable `break;` statements it also does not complain about the fallthrough that occurs. It should be noted that the two aren't related anyway.
I'm not able to test this on the build env on short notice, but the problem seems to lie with my OS X environment anyway.

**Comment by sephie** [comment]
@nzakas , both environments build from the same package.json and I've verified that the same version was used.
I've switched to ESLint v1 for now, since we're going to use it on production.

**Comment by ilyavolodin** [comment]
@sephie Is the same issue happening with ESLint v1.10.3?

**Comment by sephie** [comment]
No it does not happen. This issue came to light when I upgraded to the v2 beta.
v1.10.3 does not complain about any of the two errors.
To be clear, this is the code as it is now, with no errors in ESLint v1:

```javascript
//(switch + case)...
  else {
    //...
    return false;
}
break;
case KEY_CODE.PLUS:
case KEY_CODE.PLUSNUMERICPAD:
            var inputElement = this;
//...
```
`break;` is unreachable and the `case` below it falls through without any error.

I hope I haven't missed anything. We're implementing Babel+ESLint and are removing JSHint+JSCS, so it's a bit messy. After finalizing that I can look at the situation in detail.

**Comment by ilyavolodin** [comment]
@sephie This is very strange. The only thing that might differ between operating systems is file traversal. Once you get a chance, could you run eslint with `--debug` flag on both machines and verify that the same files are being processed?

**Comment by nzakas** [comment]
@sephie any chance you can share the full switch statement?

**Comment by eslintbot** [comment]
Hi @sephie, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by sephie** [comment]
Sorry for the delay, but here's the code I referenced in relation to the fallthrough. I've still omitted the contents of the `if` blocks for clarity.
There are multiple `switch` statements used similar to this one, but this is the one from which I took the errors.

This gives no errors with our current ESLint config (ESLint v1.10.3). I see ESLint v2 has been released. Congrats! I'll see to upgrading to the new version to see what the current state is.

```javaScript
component.$input.on('keydown', function(e) {
        switch (e.keyCode) {
            case KEY_CODE.ENTER:
                if (component.selectedIndex >= 0) {
                    //...
                    return false;
                }
                else if (component.options.length && component.options[0].isExact && !component.altSuggestionsIsOpen) {
                    //...
                    return false;
                }
                else if (component.request && component.request.state() === 'pending') {
                    //...
                    return false;
                }
                else {
                    //...
                    return false;
                }
                break;
            case KEY_CODE.PLUS:
            case KEY_CODE.PLUSNUMERICPAD:
                if (component.selectedIndex >= 0) {
                    //...
                    return false;
                }
                else if (component.options.length && component.options[0].isExact && !component.altSuggestionsIsOpen) {
                    //...
                    return false;
                }
                else if (component.request.state() === 'pending') {
                    //...
                    return false;
                }
                else {
                    //...
                    return false;
                }
                break;
        }
    });

**Comment by sephie** [comment]
Using ESLint 2.1 with parser babel-eslint. Parent directory also contains a `.eslintrc`:
```json
{
    "extends": "eslint:recommended",
    "env": {
        "node": true
    }
}
```

With the code in my previous comment, after upgrading I get one type of error (two times):
`warning  Unreachable code  no-unreachable` for two `break;` statements that follow:
```javaScript
else{
  //...
  return false;
}
break; //unreachable
```
My observations:
- `eslint-recommended` settings have both `no-unreachable` and `no-fallthrough` flags enabled by default.
- ESLint v1: unreachable `break;` & fallthrough `case:` both not caught.
- ESLint v2: unreachable `break;` caught. fallthrough `case:` not caught.
- Desired behaviour for any version of ESLint is to catch both types.
- tested with `babel-eslint` parser and unspecified (default) parser. 

I hope this helps

**Comment by nzakas** [comment]
This looks like the correct behavior. Both `break` statements are unreachable. Fallthroughs are not flagged when the cases are directly next to each other. Is there something else you're concerned about?

**Comment by sephie** [comment]
Ah, I didn't know about no-fallthrough and having cases right below each other. Thanks! It should be fine then.

**Comment by nzakas** [comment]
Sure thing. For reference, here's the docs:  http://eslint.org/docs/rules/no-fallthrough
